core/src/Cabana_MemberTypes.hpp

```diff
@@ -74,6 +74,54 @@ struct MemberTypeAtIndex<M, MemberTypes<Types...>>
     using type = typename MemberTypeAtIndexImpl<M, Types...>::type;
 };
 
+//---------------------------------------------------------------------------//
+/*!
+  \class CheckMemberTypes
+  \brief Check that member types are valied.
```

Comment:
s/valied/valid/

Also from that description I cannot tell how this differs from `is_member_type`